### PR TITLE
May 2017 roadmap

### PR DESCRIPTION
**Pull request by raphlinus**
We've been talking about where development and features fit on a
roadmap for a while, so now seems like a good time to formalize it.

This commit introduces a roadmap as of May 2017, at least roughly
capturing our thinking. It also updates the main README a bit, to
refer to the roadmap and also provide more up-to-date information
about the status of front-ends.

Many thanks to @cmyr for the first draft!

### CODE REVIEW

**Comment by raphlinus** [comment]
Open for discussion of what the roadmap should look like. Did I miss an important milestone? Something obviously out of order? A major intent is to make it easier for contributors to figure out what to work on.

File: doc/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+### next/alpha/towards a useable editor (summer 2017?)
+The immediate goal is to get xi to a point where it is a suitable choice for basic text editing, and to enable development of basic plugins.
+
+- [ ] automatic linewrap (#184)
+- [ ] horizontal scrolling (https://github.com/google/xi-mac/issues/4)
+- [ ] simple find/replace (https://github.com/google/xi-mac/issues/9, #150)
+- [ ] automatic syntax highlighting (inc. automatic syntax detection) (this comes down to improved plugin support)
+- [x] warn on close with unsaved changes (#174)
+- [ ] preliminary plugin support (#189)
+- [ ] Goto-line ? (#190)
```
Comment (cmyr) [inline doc/roadmap.md]: this is now complete, as of #192 (thanks @cbrewster).

File: doc/roadmap.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+- [ ] modal editing (vi mode) (#93)
+- [ ] rich text (similar document model as Markdown)
+- [ ] some form of workspace, or navigation through multiple files (#181)
+
+### deep future
+(who knows)
+
+- [ ] collaborative editing
+- [ ] remote sessions
```
Comment (cmyr) [inline doc/roadmap.md]: are these two items superseded by "multi device editing"?

Comment (raphlinus) [inline doc/roadmap.md]: No, I consider them separate. Collaborative editing is multiple users editing at the same time, so among other things would include displaying cursors of the other users.

**Comment by raphlinus** [comment]
The issue links didn't render as I expected. Maybe that's a sign this should be an issue rather than a document? I'm sure there are advantages and disadvantages to each. In any case, those should be fixed if this gets committed.

**Comment by trishume** [comment]
IMO it should be an issue. Easier commenting, editing, and you can actually click the check boxes and drag them around.

**Comment by raphlinus** [comment]
Will change to an issue. Also (per IRC discussion) will add entries for testing, auto-indent plugin, and making sure xi-lang gets into the "eventually" discussion.

**Comment by rkusa** [comment]
The README or ROADMAP could also mention the existence of `help-wanted` labels.

**Comment by lqd** [comment]
should this point to #252 instead of this PR ?

**Comment by cmyr** [comment]
@lqd yes, thanks!

**Pull request merged**